rectools/models/nn/transformers/lightning.py

```diff
@@ -83,6 +85,7 @@ def __init__(
         self.data_preparator = data_preparator
         self.lr = lr
         self.loss = loss
+        self.loss_calculator, self.requires_negatives = self.get_loss_calculator()
```

Comment:
Can we please split `loss_calculator` logic and `requires_negatives` logic?
We can now finally make it all clean with negatives requirement. We have a very tricky solution in data preparator initialization: https://github.com/MobileTeleSystems/RecTools/blob/f1583c5127de16760fef237733181c919f2a3a64/rectools/models/nn/transformers/base.py#L320C16-L320C78
We just hardcode "softmax" loss check there and it is not good.

In our new decision we can make it in the correct way all together: if we implement `requires_negatives(loss: str) -> bool` as a static method of `LightningModuleBase` then we can call it in data preparator initialization as `requires_negatives = self.lightning_module_type.requires_negatives(self.loss)`

Could we please do this as a last thing here? :)

Reply:
That's a great point.
Fixed.